app/src/main/kotlin/cz/covid19cz/erouska/exposurenotifications/worker/SelfCheckerWorker.kt

```diff
@@ -33,12 +36,25 @@ class SelfCheckerWorker(
                 LocalNotificationsHelper.showOutdatedDataNotification(context)
             }
 
-            if (InAppUpdateHelper.isObsolete()) {
-                InAppUpdateHelper.checkForAppUpdateAndNotify(context)
+            if (isEligibleToShowAppUpdateNotification()) {
+                InAppUpdateHelper.checkForAppUpdateAndNotify(context) {
+                    prefs.setLastTimeAppUpdateNotificationShown(System.currentTimeMillis())
+                    prefs.setLastVersionAppUpdateNotificationShown(BuildConfig.VERSION_CODE)
+                }
             }
         }
-
         return Result.success()
     }
 
+    /**
+     * App is eligible to show local notification about app update.
+     * Returns true if the version is new or if the notification was shown last time before at least x-days (configurable via RC).
+     */
+    private fun isEligibleToShowAppUpdateNotification(): Boolean {
+        return prefs.getLastVersionAppUpdateNotificationShown() != BuildConfig.VERSION_CODE ||
```

Comment:
Tuhle podmínku nechápu. Proč se porovnává versionCode? To se pro jeden versionCode zobrazí notifikace jenom jednou?

Reply:
Porovnává se `versionCode`, abychom v každé verzi aplikace měl ihned možnost force update.
Když by tam ta podmínka nebyla, tak pokud bysme chtěli např. verzi 10 force updatnout na verzi 11 a ihned po jejím vydáním bysme si uvědomili, že je problémová a vydali bugfixovou verzi 12, tak by aplikace ve verzi 11 force update notifikaci neukázala dříve než po uplynutí dnů specifikovaných přes `forceAppUpdatePeriodDays`